Readme.md

```diff
@@ -0,0 +1,369 @@
+openssl req -new -x509 -key ca/key -out ca/cert -config $CONF
+# Server 
+openssl genpkey -algorithm ed25519 > server/key
+openssl req -new -key server/key -out server/csr -config $CONF
+openssl x509 -req -in server/csr -extfile $CONF -extensions server_ext -out server/cert -CA ca/cert -CAkey ca/key 
+# Client
+openssl genpkey -algorithm ed25519 > client/key
+openssl req -new -key server/key -out client/csr -config $CONF
+openssl x509 -req -in server/csr -extfile $CONF -extensions client_ext -out client/cert -CA ca/cert -CAkey ca/key 
+```
+
+<details>
+<summary> Click to view openssl config </summary>
+
+```sh 
+[ req ]
+default_bits        = 4096
+default_md          = sha512
+distinguished_name  = req_distinguished_name
+
+[ req_distinguished_name ]
+countryName                     = Country Name (2 letter code)
+stateOrProvinceName             = State or Province Name
+localityName                    = Locality Name
+0.organizationName              = Organization Name
+organizationalUnitName          = Organizational Unit Name
+commonName                      = Common Name
+emailAddress                    = Email Address
+countryName_default = CA 
+stateOrProvinceName_default = BC 
+localityName_default = Vancouver
+
+[ client_ext ]
+basicConstraints        = CA:FALSE
+subjectKeyIdentifier    = hash
+authorityKeyIdentifier  = keyid, issuer
+keyUsage                = critical, nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage        = clientAuth, emailProtection
+
+[ server_ext ]
+basicConstraints        = CA:FALSE
+subjectKeyIdentifier    = hash
+authorityKeyIdentifier  = keyid, issuer:always
+keyUsage                = critical, digitalSignature, keyEncipherment
+extendedKeyUsage        = serverAuth
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = localhost
+DNS.2 = 127.0.0.1
+DNS.3 = ::1
+```
+</details>
+
+Application leverages [rustls](https://github.com/rustls/rustls) for crypto needs, which had been [audited by Cure53](https://github.com/rustls/rustls/blob/master/audit/TLS-01-report.pdf) on behalf of CNCF in 2020. 
+
+## gRPC 
+
+[tonic](https://github.com/hyperium/tonic) is default gRPC server/client library in Rust ecosystem.
+
+<details>
+<summary> Click to view protobuf </summary>
+
+```protobuf
+syntax = "proto3";
+
+package runner;
+
+service JobRunner {
+  rpc Start(JobRequest) returns (JobId);
+  rpc Stop(JobId) returns (Ack);
+  rpc Status(JobId) returns (JobStatus);
+  rpc Output(JobId) returns (stream LogMessage);
+}
+
+message JobRequest {
+  message CpuControl { uint32 cpu_weight = 1; }
+
+  message MemControl {
+    uint64 mem_high = 1;
+    uint64 mem_max = 2;
+  }
+
+  message IoControl {
+    uint64 rbps_max = 1;
+    uint64 wbps_max = 2;
+  }
+
+  string executable = 1;
+  optional CpuControl cpu_control = 2;
+  optional MemControl mem_control = 3;
+  optional IoControl io_control = 4;
+  repeated string args = 5;
+}
+
+message Ack {}
+
+message JobId { bytes jobid = 1; }
```

Comment:
What is used as a job ID?

Reply:
Intention is to use [uuid](https://docs.rs/uuid/latest/uuid/), added comment.